geopy/geocoders/arcgis.py

```diff
@@ -118,14 +118,27 @@ def geocode(self, query, exactly_one=True, timeout=None):
         :param bool exactly_one: Return one result or a list of results, if
             available.
 
+        :param out_fields:
+        :type out_fields: A list of output fields to be returned in the
+            attributes field of the raw data. This can be either a python
+            list/tuple of fields or a comma-separated string.
+
         :param int timeout: Time, in seconds, to wait for the geocoding service
             to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
             exception. Set this only if you wish to override, on this call
             only, the value set during the geocoder's initialization.
         """
-        params = {'text': query, 'f': 'json'}
+        params = {'singleLine': query, 'f': 'json'}
         if exactly_one is True:
             params['maxLocations'] = 1
+        if out_fields is not None:
+            # Python 2/3 compatibility
+            if 'basestring' not in globals():
+                basestring = str
+            if isinstance(out_fields, basestring):
```

Comment:
You can use `geopy.compat.string_compare` here: `if isinstance(out_fields, string_compare):`